### ISSUE TITLE
Tab-completing an environment variable raises an exception

### ISSUE THREAD

**Issue by qwenger** [xonsh/xonsh#3694]
Tab-completing environment variables (beginning with a "$", e.g. $XONSH_COLOR_STYLE) raises the exception reproduced below.

## xonfig

<details>

```
+------------------+---------------------+
| xonsh            | 0.9.19              |
| Git SHA          | 74450cb5            |
| Commit Date      | Aug 9 13:27:55 2020 |
| Python           | 3.8.5               |
| PLY              | 3.11                |
| have readline    | True                |
| prompt toolkit   | 3.0.5               |
| shell type       | prompt_toolkit      |
| pygments         | 2.6.1               |
| on posix         | True                |
| on linux         | True                |
| distro           | arch                |
| on darwin        | False               |
| on windows       | False               |
| on cygwin        | False               |
| on msys2         | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
+------------------+---------------------+

```

</details>

## Expected Behavior

The autocompletion should run.

## Current Behavior

An exception is raised.

### Traceback (if applicable)

<details>

```
Unhandled exception in event loop:
  File "/usr/lib/python3.8/asyncio/events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 653, in read_from_input
    self.key_processor.process_keys()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/key_binding/key_processor.py", line 274, in process_keys
    self._process_coroutine.send(key_press)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/key_binding/key_processor.py", line 186, in _process
    self._call_handler(matches[-1], key_sequence=buffer[:])
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/key_binding/key_processor.py", line 329, in _call_handler
    handler.call(event)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/key_binding/key_bindings.py", line 101, in call
    self.handler(event)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/shortcuts/prompt.py", line 789, in _complete_like_readline
    display_completions_like_readline(event)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/key_binding/bindings/completion.py", line 65, in display_completions_like_readline
    completions = list(b.completer.get_completions(b.document, complete_event))
  File "/usr/lib/python3.8/site-packages/xonsh/ptk_shell/completer.py", line 51, in get_completions
    completions, l = self.completer.complete(
  File "/usr/lib/python3.8/site-packages/xonsh/__amalgam__.py", line 125, in complete
    out = func(prefix, line, begidx, endidx, ctx)
  File "/usr/lib/python3.8/site-packages/xonsh/completers/__amalgam__.py", line 1421, in complete_path
    paths.update(filter(filtfunc, _env(prefix)))
  File "/usr/lib/python3.8/site-packages/xonsh/completers/__amalgam__.py", line 1189, in _env
    return {
  File "/usr/lib/python3.8/site-packages/xonsh/completers/__amalgam__.py", line 1190, in <setcomp>
    "$" + k for k in builtins.__xonsh__.env if get_filter_function()(k, key)
  File "/usr/lib/python3.8/site-packages/xonsh/completers/__amalgam__.py", line 644, in _filter_normal
    return s.startswith(x)

Exception 're.Pattern' object has no attribute 'startswith'
```

</details>

## Steps to Reproduce

1. type "$X"
2. press TAB

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by laloch** [comment]
Hi @qwenger, thanks for reporting!
This should already be fixed in #3679.

**Comment by qwenger** [comment]
Hi @laloch, you're right, the master version works. Thank you!
